src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -1779,6 +1779,8 @@ _public_ void sd_bus_close(sd_bus *bus) {
 _public_ sd_bus *sd_bus_close_unref(sd_bus *bus) {
         if (!bus)
                 return NULL;
+        if (bus_pid_changed(bus))
```

Comment:
so i think this makes sense. After all sd_bus_close() already has something like this, hence this just normalizes behaviour here.

However, there's one thing missing: sd_bus_unref() itself should operate like this too. It's not covered righ tnow because it is implemented using DEFINE_PUBLIC_TRIVIAL_REF_UNREF_FUNC(). We probably should implement that manually, and add the very same check there too, so that all destructors are covered, not just some.